src/data/sum/order.lean

```diff
@@ -524,3 +525,49 @@ end,
   (sum_lex_dual_antidistrib α β).symm (inr (to_dual a)) = to_dual (inl a) := rfl
 
 end order_iso
+
+variable [has_le α]
+
+namespace with_bot
+
+/-- `with_bot α` is order-isomorphic to `punit ⊕ₗ α`, by sending `⊥` to `punit.star` and `↑a` to
+`a`. -/
+def order_iso_punit_sum_lex : with_bot α ≃o punit ⊕ₗ α :=
```

Comment:
I would have expected `order_iso.punit_sum_lex` or something like that.

Reply:
I think it would be best if these definitions had `with_bot`/`with_top` in either the name or namespace. Maybe `order_iso.with_bot_order_iso_punit_sum_lex` to sort of match `equiv.option_equiv_sum_punit?